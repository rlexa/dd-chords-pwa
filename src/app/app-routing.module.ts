import {NgModule} from '@angular/core';
import {RouterModule, Routes} from '@angular/router';

const routes: Routes = [
  {path: 'ui', loadChildren: () => import('src/app/module/dashboard/routed-dashboard.module').then((m) => m.RoutedDashboardModule)},
  {path: '', redirectTo: 'ui', pathMatch: 'full'},
  {path: '*', redirectTo: ''},
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
