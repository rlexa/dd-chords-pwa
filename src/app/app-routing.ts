export const routeShared = 'shared';
export const routeUi = 'ui';
