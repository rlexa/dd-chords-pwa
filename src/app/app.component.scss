:host {
  position: absolute;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  min-width: 300px;
  overflow-x: auto;
}
