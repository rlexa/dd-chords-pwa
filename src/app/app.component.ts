import {ChangeDetectionStrategy, Component} from '@angular/core';
import {RouterModule} from '@angular/router';
import {ServiceWorkerWatchDirective} from './shared/service-worker-watch';
import {TrackImportWatchDirective} from './shared/track-import-watch/track-import-watch.directive';

@Component({
  selector: 'dd-chords-app',
  template: `<router-outlet ddChordsServiceWorkerWatch ddChordsTrackWatchImport />`,
  styles: [
    `
      :host {
        position: absolute;
        left: 0px;
        top: 0px;
        width: 100%;
        height: 100%;
        min-width: 300px;
        overflow-x: auto;
      }
    `,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [RouterModule, ServiceWorkerWatchDirective, TrackImportWatchDirective],
})
export class AppComponent {}
