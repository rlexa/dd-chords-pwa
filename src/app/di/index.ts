export * from './active';
export * from './current-performer.di';
export * from './current-performers.di';
export * from './current-router-params.di';
export * from './current-track-count.di';
export * from './current-track-hashes.di';
export * from './current-tracks.di';
export * from './music-idb-live.di';
export * from './music-idb.di';
export * from './online.di';
export * from './performer-filter-favorites.di';
export * from './performers-filter-part-query.di';
export * from './performers-filter.di';
export * from './tracks-filter-favorites.di';
export * from './tracks-filter-part-performer.di';
export * from './tracks-filter-part-query.di';
export * from './tracks-filter.di';
