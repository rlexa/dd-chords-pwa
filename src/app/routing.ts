export const RouteInfo = 'info';
export const RouteShared = 'shared';
export const RouteTracks = 'tracks';
export const RouteUi = 'ui';

export const RouteParamIdTrack = 'idTrack';

export const QueryParamTrackId = 'id';
