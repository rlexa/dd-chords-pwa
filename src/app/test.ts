import {Provider, Type} from '@angular/core';
import {Mock} from 'ts-mockery';

export const mockWith = <T>(provide: Type<T>, mock: T) => ({provide, useFactory: () => mock}) as Provider;

export const mockAll = <T>(provide: Type<T>) => mockWith(provide, Mock.all<T>());
