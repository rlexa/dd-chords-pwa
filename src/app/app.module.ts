import {HttpClientModule} from '@angular/common/http';
import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {ServiceWorkerModule, SwUpdate} from '@angular/service-worker';
import {from, merge} from 'rxjs';
import {delay, switchMap, tap} from 'rxjs/operators';
import {environment} from '../environments/environment';
import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';
import {DiCommonModule} from './module/common/di-common';
import {LoggerService} from './module/common/logger';
import {RoutingService} from './module/common/routing/routing-service';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    HttpClientModule,
    AppRoutingModule,
    DiCommonModule,
    ServiceWorkerModule.register('ngsw-worker.js', {enabled: environment.production}),
  ],
  bootstrap: [AppComponent],
})
export class AppModule {
  constructor(routingService: RoutingService, swUpdate: SwUpdate, loggerService: LoggerService) {
    merge(
      swUpdate.available.pipe(tap((event) => loggerService.log(`Detected update, reloading soon...`, event))),
      swUpdate.unrecoverable.pipe(tap((event) => loggerService.error(`Detected unrecoverable state "${event.reason}", reloading soon...`))),
    )
      .pipe(
        delay(5000),
        switchMap(() => from(swUpdate.activateUpdate())),
      )
      .subscribe(() => document.location.reload());
  }
}
