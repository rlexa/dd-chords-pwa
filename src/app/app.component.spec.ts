import {ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AppComponent} from './app.component';
import {detectChanges, overrideForChangeDetection} from './test';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent],
    })
      .overrideComponent(AppComponent, overrideForChangeDetection)
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    detectChanges(fixture);
  });

  it('has instance', () => expect(fixture.componentInstance).toBeTruthy());
  it('renders', () => expect(fixture).toMatchSnapshot());
});
