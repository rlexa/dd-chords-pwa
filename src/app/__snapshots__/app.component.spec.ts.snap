// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent renders 1`] = `
<dd-chords-app>
  <router-outlet />
</dd-chords-app>
`;
